test/test_autograd.py

```diff
@@ -4922,6 +4922,30 @@ def fn(*inputs):
             setattr(TestAutogradDeviceType, test_name, do_test)
 
 class TestAutogradComplex(TestCase):
+    @skipIfNoLapack
+    def test_complex_cholesky(self):
```

Comment:
I think we should run the existing `test_cholesky` for complex as well instead of adding a new test. `TestAutogradComplex` was added to temporarily add some tests for complex while `gradcheck` for complex wasn't implemented.

Reply:
I've merged the complex and real variants. Since `TestAutograd` is not instantiated with `instantiate_device_type_tests`, I added dtype parameterization inside `test_cholesky` instead of `@dtypes`.